partiql-plan/src/main/kotlin/org/partiql/plan/v1/rex/Rex.kt

```diff
@@ -0,0 +1,12 @@
+package org.partiql.plan.v1.rex
+
+import org.partiql.plan.v1.Node
+import org.partiql.types.PType
+
+/**
+ * TODO DOCUMENTATION
+ */
+public interface Rex : Node {
+
+    public fun getType(): PType
```

Comment:
nice, I like that the type is on the same level as the `Rex` implementations. Was pretty annoying with the current plan APIs when the `Rex.Op`s couldn't directly reference their `type` field.

Reply:
This is one of the primary goals, on top of being able to have internal implementations of these operators for the typers without having to transform across domains. I am making an effort to flatten things for ease-of-use.